bindings/mqtt/mqtt.go

```diff
@@ -122,13 +266,39 @@ func (m *MQTT) connect(clientID string, uri *url.URL) (mqtt.Client, error) {
 	return client, nil
 }
 
-func (m *MQTT) createClientOptions(clientID string, uri *url.URL) *mqtt.ClientOptions {
+func (m *MQTT) newTLSConfig() *tls.Config {
+	tlsConfig := new(tls.Config)
+
+	if m.metadata.clientCert != "" && m.metadata.clientKey != "" {
+		cert, err := tls.X509KeyPair([]byte(m.metadata.clientCert), []byte(m.metadata.clientKey))
+		if err != nil {
+			m.logger.Warnf("unable to load client certificate and key pair. Err: %v", err)
+
+			return tlsConfig
+		}
+		tlsConfig.Certificates = []tls.Certificate{cert}
+	}
+
+	if m.metadata.caCert != "" {
+		tlsConfig.RootCAs = x509.NewCertPool()
+		if ok := tlsConfig.RootCAs.AppendCertsFromPEM([]byte(m.metadata.caCert)); !ok {
+			m.logger.Warnf("unable to load ca certificate.")
+		}
+	}
+
+	return tlsConfig
+}
+
+func (m *MQTT) createClientOptions(uri *url.URL, clientID string) *mqtt.ClientOptions {
 	opts := mqtt.NewClientOptions()
-	opts.AddBroker(fmt.Sprintf("tcp://%s", uri.Host))
+	opts.SetClientID(clientID)
+	opts.SetCleanSession(m.metadata.cleanSession)
+	opts.AddBroker(uri.Scheme + "://" + uri.Host)
```

Comment:
@Hugome is this a breaking change? The current docs/old code have the schema as `mqtt[s]` and no matter what it is forced to `tcp`. Would it make sense to replace `mqtts` with `tcps` and `mqtt` with `tcp`? If we do something like that we can preserve backward compatability. WDYT?

Reply:
Perhaps based on this it should be `ssl` instead of `tcps` and we have it wrong in the pubsub version...  Not sure.

https://pkg.go.dev/github.com/eclipse/paho.mqtt.golang#ClientOptions.AddBroker

Reply:
Oh yes you're right this is weird.
Maybe just the pubsub docs has issue ? (+1 for `tcp://` and `tcps://`)

Reply:
(Updated on the docs for `ssl://` https://github.com/dapr/docs/pull/1328/commits/cbc74878fa85e3b4e15aa82fe9f65e71efec1da3)

Reply:
So, is it a breaking change or not?

Reply:
Currently yes.  What I think we (or @Hugome) needs to add is a schema coalesce to convert the previous schemes to what the MQTT client is expecting (ex. `mqtt://` -> `tcp://`, `mqtts` -> `ssl://`, etc.)

Reply:
To move this along, I added the following code to coalesce the scheme value
```golang
// URL scheme backward compatibility
scheme := uri.Scheme
switch scheme {
case "mqtt":
	scheme = "tcp"
case "mqtts", "tcps":
	scheme = "ssl"
}
opts.AddBroker(scheme + "://" + uri.Host)
```